mods/env_sounds/init.lua

```diff
@@ -0,0 +1,97 @@
+-- Parameters
+
+local radius = 8 -- Water node search radius around player
+
+-- End of parameters
+
+
+local handles = {}
+local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
+
+
+-- Update sound for player
+
+local function update_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for water nodes in radius around player
+	local ppos = player:get_pos()
+	local areamin = vector.subtract(ppos, radius)
+	local areamax = vector.add(ppos, radius)
+	local wpos, num
+	if river_source_sounds then
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_source",
+				"default:river_water_flowing"
+			}
+		)
+	else
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_flowing"
+			}
+		)
+	end
+	-- Total number of waters in radius
+	local waters = #wpos
+	if waters == 0 then
+		return
+	end
+
+	-- Find average position of water positions
+	local wposav = vector.new()
+	for i, pos in ipairs(wpos) do
+		wposav = vector.add(wposav, pos)
+	end
+	wposav = vector.divide(wposav, waters)
+	-- Play sound
+	local handle = minetest.sound_play(
+		"env_sounds_water",
+		{
+			pos = wposav,
+			to_player = player_name,
+			gain = math.min(0.04 + waters * 0.004, 0.4),
+			max_hear_distance = 32,
```

Comment:
shouldn't this be (roughly) `radius` so the sound stops playing if the player quickly leaves the area?

Reply:
I may have misunderstood you and you may already know this, but, this parameter doesn't control the maximum distance a sound can be heard once it starts playing. When a sound starts playing at a position, all clients within this distance of the sound are sent a 'start sound' instruction.

Either way, i possibly agree, the average water node position will be within 'radius' of the player, i just have to be careful of potential square versus circular radii.

Reply:
This parameter could even be omitted. As far as I can see, it is not used if `to_player` isn't empty (see https://github.com/minetest/minetest/blob/fae6242d4ef6ca34d131054484acf5efc2fcba70/src/server.cpp#L1975-L1987 and https://github.com/minetest/minetest/blob/fae6242d4ef6ca34d131054484acf5efc2fcba70/src/server.cpp#L2002).

Reply:
If it's unused by the engine it should indeed be omitted.

Reply:
Thanks, will omit.